### ISSUE TITLE
Don't report retryErrors

### ISSUE THREAD

**Issue by shepazon** [awslabs/aws-sdk-swift#559]
## Description
Currently, when errors occur, the resulting error includes a layer marked as a `retryError`. This should not be included, as retries are handled internally by the SDK.

## Context
For example, when a call to `getSecretValue()` is made with a secret name that doesn't exist, a `resourceNotFoundException` should be thrown. Instead, what's thrown is a nested error as follows:

```
(ClientRuntime.ClientError.retryError(ClientRuntime.SdkError<AWSSecretsManager.GetSecretValueOutputError>.service(AWSSecretsManager.GetSecretValueOutputError.resourceNotFoundException(AWSSecretsManager.ResourceNotFoundException(_headers: Optional(Content-Length: 99, 
x-amzn-RequestId: 1f81fd3f-88fa-46e9-a4b4-7561ea7210b8, 
Date: Mon, 21 Mar 2022 19:48:30 GMT, 
Content-Type: application/x-amz-json-1.1), _statusCode: Optional(http_status_400), _message: Optional("Secrets Manager can\'t find the specified secret."), _requestID: Optional("1f81fd3f-88fa-46e9-a4b4-7561ea7210b8"), _retryable: false, _isThrottling: false, _type: ClientRuntime.ErrorType.client, message: Optional("Secrets Manager can\'t find the specified secret."))), 
Status Code: http_status_400 
 Content-Length: 99, 
x-amzn-RequestId: 1f81fd3f-88fa-46e9-a4b4-7561ea7210b8, 
Date: Mon, 21 Mar 2022 19:48:30 GMT, 
Content-Type: application/x-amz-json-1.1)), nil)
```

Digging the useful error out of this is a bit of work that shouldn't be needed, since the top error seems to be something other than what the true error is.

## Environment

* AWS Swift SDK version used: 0.2.2
* Operating System and version (iOS/macOS/Linux): macOS 12.3
* Xcode version (if applicable): 13.2.1

## Reproduction Steps
This seems to happen with most or all SDK calls.

**Comment by ganeshnj** [comment]
We need to look at holistically, how to improve the error handling experience better for end user.

**Comment by royjit** [comment]
It is hard to parse through error right now. Getting a specific service error from the error thrown currently required to traverse multiple layers:

```
// Code to check if api returned userNotConfirmedException

if let cognitoError = serviceError as? SdkError<InitiateAuthOutputError>,
                      case .client(let clientError, _) = cognitoError,
                      case .retryError(let retryError) = clientError,
                      let cognitoServiceError = retryError as? SdkError<InitiateAuthOutputError>,
                      case .service(let internalServiceError, _) = cognitoServiceError,
                      case .userNotConfirmedException = internalServiceError {
                return true
}
```

You can see from the above code  `SdkError<InitiateAuthOutputError>` is wrapped internally again inside retryError. The above code should be simple as:

```
if let cognitoError = serviceError as? SdkError<InitiateAuthOutputError>,
               case .service(let internalServiceError, _) = cognitoError,
               case .userNotConfirmedException = internalServiceError {
           return true
}
```

**Comment by alteredtech** [comment]
While this might just be me missing a huge part of this sdk or swift in general. I believe the OutputError should just have .error and .error_message. Just anything that will allow access to the actual error.

**Comment by jbelkins** [comment]
Done & shipped in latest release.  Closing.
